src/huggingface_hub/hf_api.py

```diff
@@ -1437,35 +1515,33 @@ def upload_file(
         doesn't require git or git-lfs to be installed.
 
         Args:
-            upload_fileth_or_fileobj (`str`, `bytes`, or `IO`):
+            path_or_fileobj (`str`, `bytes`, or `IO`):
                 Path to a file on the local machine or binary data stream / fileobj / buffer.
-
             path_in_repo (`str`):
-                Relative filepath in the repo, for example: :obj:`"checkpoints/1fec34a/weights.bin"`
-
+                Relative filepath in the repo, for example: `"checkpoints/1fec34a/weights.bin"`
             repo_id (`str`):
-                The repository to which the file will be uploaded, for example: :obj:`"username/custom_transformers"`
-
-            token (`str`):
+                The repository to which the file will be uploaded, for example: `"username/custom_transformers"`
+            token (`str`, *optional*):
                 Authentication token, obtained with :function:`HfApi.login` method. Will default to the stored token.
-
-            repo_type (`str`, Optional):
-                Set to :obj:`"dataset"` or :obj:`"space"` if uploading to a dataset or space, :obj:`None` or :obj:`"model"` if uploading to a model. Default is :obj:`None`.
-
-            revision (`str`, Optional):
-                The git revision to commit from. Defaults to the :obj:`"main"` branch.
-
-            identical_ok (`bool`, defaults to `True`):
+            repo_type (`str`, *optional*):
+                Set to `"dataset"` or `"space"` if uploading to a dataset or space, `None` or `"model"` if uploading to a model. Default is `None`.
+            revision (`str`, *optional*):
+                The git revision to commit from. Defaults to the `"main"` branch.
+            identical_ok (`bool`, *optional*, defaults to `True`):
                 When set to false, will raise an HTTPError when the file you're trying to upload already exists on the hub
                 and its content did not change.
 
         Returns:
             `str`: The URL to visualize the uploaded file on the hub
 
-        Raises:
-            :class:`ValueError`: if some parameter value is invalid
+        <Tip>
+
+        Raises the following errors:
 
-            :class:`requests.HTTPError`: if the HuggingFace API returned an error
+            - `HTTPError` if the HuggingFace API returned an error
+            - `ValueError` if some parameter value is invalid
```

Comment:
and here. These should be links to the corresponding docs.

Reply:
Great idea, much better this way!